Library/Homebrew/dev-cmd/audit.rb

```diff
@@ -910,13 +913,16 @@ def audit_revision_and_version_scheme
           newest_committed_version ||= previous_version
           newest_committed_checksum ||= previous_checksum
           newest_committed_revision ||= previous_revision
+          newest_committed_url ||= stable.url
         end
 
         break if previous_version && current_version != previous_version
       end
 
       if current_version == previous_version &&
-         current_checksum != newest_committed_checksum
+         current_checksum != newest_committed_checksum &&
+         (newest_committed_url.start_with?(PyPI::PYTHONHOSTED_URL_PREFIX) ==
+          current_url.start_with?(PyPI::PYTHONHOSTED_URL_PREFIX))
```

Comment:
I haven't really looked at the rest of the code but this doesn't seem to be anything specific to PyPI. If the URL changes: the checksum can change, too.